src/extensions/registries/page-registry.ts

```diff
@@ -59,24 +60,27 @@ export function getExtensionPageUrl<P extends object>({ extensionId, pageId = ""
   return extPageRoutePath;
 }
 
-export class PageRegistry extends BaseRegistry<PageRegistration, RegisteredPage> {
+export class PageRegistry extends BaseRegistry<RegisteredPage> {
   @action
-  add(items: PageRegistration[], ext: LensExtension) {
-    let registeredPages: RegisteredPage[] = [];
+  add(items: PageRegistration | PageRegistration[], ext: LensExtension) {
+    const itemArray = Array.isArray(items) ? items : [items];
     try {
-      registeredPages = items.map(page => ({
+      const pages = itemArray.map(page => ({
         ...page,
         extensionId: ext.name,
         routePath: getExtensionPageUrl({ extensionId: ext.name, pageId: page.id ?? page.routePath }),
       }));
+
+      return super.add(pages);
     } catch (err) {
       logger.error(`[EXTENSION]: page-registration failed`, {
         items,
         extension: ext,
         error: String(err),
       });
+
+      return noop;
     }
-    return super.add(registeredPages);
```

Comment:
can't you keep calling this and let BaseRegistry handle the noop situation? Seems cleaner the way it was. Or am I missing the point?

Reply:
This was mostly done so that I could remove the above `let` and make it a `const`

Reply:
I agree with @jim-docker , is `const` really that important here?